File: hw/viteri/u3/HW30ObserverPatternandStrategy/HW30Strategy/src/main/java/model/SortingContext.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

import controller.SortingStrategy;

/**
 *
 * @author Alexis Viteri DCO-ESPE
 */
public class SortingContext {
    private SortingStrategy ss;
    
    public void sort(int data[]){
        int size = data.length;
        ss = setSortStrategy(size);
        ss.sort(data);
    }
    
    public SortingStrategy setSortStrategy(int size){
        if(size > 0 && size < 30){
            System.out.println("\nSorting with the bubble algorithm");
            return new BubbleSort();
            
        } else if(size >= 30 && size < 100){
            System.out.println("\nSorting with the insetion algorithm");
            return new InsertionSort();
            
        }else if(size >= 100){
            System.out.println("\nSorting with the quick algorithm");
            return new QuickSort();
            
        }
        return null;
    }
}

File: exams/villagomez/u3/examExercise/controller/SortingContext.js
import { BubbleSort } from '../model/BubbleSort.js';
import { InsertionSort } from '../model/InsertionSort.js';
import { QuickSort } from '../model/QuickSort.js';

export class SortingContext {
    constructor() {
        this.ss = null;
    }

    sort(data) {
        let size = data.length;
        this.ss = this.setSortStrategy(size);
        return this.ss.sort(data);
    }

    setSortStrategy(n) {
        if (n >= 2 && n <= 5) {
            return new BubbleSort();
        } else if (n >= 6 && n <= 10) {
            return new InsertionSort();
        } else if (n > 10) {
            return new QuickSort();
        }
    }
}
